op-node/rollup/derive/mint_token.go

```diff
@@ -0,0 +1,87 @@
+package derive
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/ethereum-optimism/optimism/op-service/solabi"
+	"github.com/kroma-network/kroma/kroma-bindings/predeploys"
+)
+
+const (
+	MintTokenFuncSignature = "mint()"
+	MintTokenLen           = 4
+)
+
+var (
+	MintTokenFuncBytes4    = crypto.Keccak256([]byte(MintTokenFuncSignature))[:4]
+	MintTokenCallerAddress = common.HexToAddress("0xdeaddeaddeaddeaddeaddeaddeaddeaddead0070")
+	MintManagerAddress     = predeploys.MintManagerAddr
+)
+
+const (
+	MintTokenTxGas = 1_000_000
+)
+
+type MintToken struct{}
+
+// Binary Format
+// +---------+--------------------------+
+// | Bytes   | Field                    |
+// +---------+--------------------------+
+// | 4       | Function signature       |
+// +---------+--------------------------+
+
+func (info *MintToken) MarshalBinary() ([]byte, error) {
+	w := bytes.NewBuffer(make([]byte, 0, MintTokenLen))
+	if err := solabi.WriteSignature(w, MintTokenFuncBytes4); err != nil {
+		return nil, err
+	}
+	return w.Bytes(), nil
+}
+
+func (info *MintToken) UnmarshalBinary(data []byte) error {
+	if len(data) != MintTokenLen {
+		return fmt.Errorf("data is unexpected length: %d", len(data))
+	}
+	reader := bytes.NewReader(data)
+
+	if _, err := solabi.ReadAndValidateSignature(reader, MintTokenFuncBytes4); err != nil {
+		return err
+	}
+	if !solabi.EmptyReader(reader) {
+		return errors.New("too many bytes")
+	}
+	return nil
+}
+
+// NewMintTokenTx creates a mint token transaction.
+func NewMintTokenTx(nonce uint64) (*types.MintTokenTx, error) {
```

Comment:
why would you introduce a new tx type when you could just have the l1 block contract be the owner of the token that can mint and have it call `mint()` when `setL1BlockValues` is called

Reply:
We have considered calling `mint()` from the `L1Block` contract as you mentioned, but we have decided to introduce a new transaction type for the reasons below:

1. We aim not to deviate from the purpose of the `L1Block` contract. We believe it's preferable for the L1Block contract to focus on setting the properties of L1Origin without performing other functions.
2. Since this transaction is not an L1 deposit action, we decided to create a new transaction type.

In other words, the introduction of a new transaction type is not due to implementation limitations but rather to align the functionality with its designated purpose.
Is this approach deviates from the Superchain's specifications or requirements?